src/Core/Services/Implementations/StripePaymentService.cs

```diff
@@ -747,6 +747,19 @@ public class StripePaymentService : IPaymentService
 
             var subResponse = await subscriptionService.UpdateAsync(sub.Id, subUpdateOptions);
 
+            var invoiceService = new InvoiceService();
+            var invoice = await invoiceService.GetAsync(subResponse?.LatestInvoiceId, new InvoiceGetOptions());
```

Comment:
Do we need to check that `subResponse` isn't null before this? `GetAsync(null, new InvoiceGetOptions())` doesn't make much sense.

Reply:
This is just an abundance of caution since the Stipe c# api isn't well documented. The `GetAsync` method will result in a exception complaining about a null id, this is effectively exactly what we do on the next line if `invoice` is returned as null.

I think it's actually helpful separating out these two error scenarios -- it is a pretty confusing way to do it, though.